diff --git a/amtl/am-algorithm.h b/amtl/am-algorithm.h
index 00ea90c..3b53718 100644
--- a/amtl/am-algorithm.h
+++ b/amtl/am-algorithm.h
@@ -34,14 +34,14 @@
 
 namespace ke {
 
-template <typename T> static inline T
-Min(const T& t1, const T& t2)
+template <typename T1, typename T2> static inline auto
+Min(const T1& t1, const T2& t2) -> decltype(t1 < t2 ? t1 : t2)
 {
     return t1 < t2 ? t1 : t2;
 }
 
-template <typename T> static inline T
-Max(const T& t1, const T& t2)
+template <typename T1, typename T2> static inline auto
+Max(const T1& t1, const T2& t2) -> decltype(t1 > t2 ? t1 : t2)
 {
     return t1 > t2 ? t1 : t2;
 }
