diff --git a/compiler/amxxpc/amxxpc.cpp b/compiler/amxxpc/amxxpc.cpp
index 895ca83..2b20291 100755
--- a/compiler/amxxpc/amxxpc.cpp
+++ b/compiler/amxxpc/amxxpc.cpp
@@ -49,6 +49,7 @@ int main(int argc, char **argv)
 #else
 # if defined(__linux__)
 	HINSTANCE lib = NULL;
+	dlopen("libm.so", RTLD_NOW | RTLD_GLOBAL);
 	if (FileExists("./amxxpc32.so"))
 		lib = dlmount("./amxxpc32.so");
 	else
@@ -68,7 +69,9 @@ int main(int argc, char **argv)
 		exit(EXIT_FAILURE);
 	}
 
-	COMPILER sc32 = (COMPILER)dlsym(lib, "Compile32");
+	COMPILER sc32 = (COMPILER)dlsym(lib, "Compile64");
+	if (!sc32)
+		sc32 = (COMPILER)dlsym(lib, "Compile32");
 	pc_printf = (PRINTF)dlsym(lib, "pc_printf");
 #endif //EMSCRIPTEN
 
@@ -121,7 +124,7 @@ int main(int argc, char **argv)
 			exit(EXIT_FAILURE);
 		}
 		ReadFileIntoPl(&pl32, fp);
-		pl32.cellsize = 4;
+		pl32.cellsize = (char)sizeof(cell);
 		fclose(fp);
 	}
 
diff --git a/compiler/libpc300/sc1.c b/compiler/libpc300/sc1.c
index f8d4061..9a492d3 100755
--- a/compiler/libpc300/sc1.c
+++ b/compiler/libpc300/sc1.c
@@ -37,7 +37,6 @@
 
 #if defined LINUX || defined __FreeBSD__ || defined __OpenBSD__ || defined __APPLE__
   #include <sclinux.h>
-  #include <prefix.h> /* from BinReloc, see www.autopackage.org */
   #include <unistd.h>
 #endif
 
